### DISCUSSION TITLE
How to set periodic BC?

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#23322] [Q&A Modules: Phase field] [answered]
Hi everyone,

I want to set the top and bottom boundary as periodic BC and left and right boundary as Neumann BC. The following shows how I set the BC, however, I found the phenomenon I got indicate that the BC setting is not correct. Can anyone give me some advice?
![image](https://user-images.githubusercontent.com/63679626/216400752-27d5f4e0-c7c7-4223-a6a7-6640a518ac66.png)

Thank you very much,
Best

**Comment by GiudGiud** [comment]
Hello

Can you please post more about what indicates the setting bc is not correct. Is there a jump in the Y direction?

Guillaume

**Reply by abc-hy** [reply]
I ran several simulation of the layer propagation, with 2 different boundary conditions and compared them. 
(1)**Case 1**
First BC I used is Neumann in all directions, I got the following first results

![image (3)](https://user-images.githubusercontent.com/63679626/216475989-58cfbf72-a446-495d-b9a7-cd243194100b.png)

Then, by using another boundary condition, which is periodic for the top and bottom boundary and Neumann for the left and right, I got this result:
![image (1)](https://user-images.githubusercontent.com/63679626/216476289-b75d4b50-12bb-49d9-bc30-136d5aa47519.png)

For the second simulation, the interface in the middle behaves differently as interfaces at the boundary.

(2) **Case 2**
Also, I ran another layer propagation simulation, First I use all Neumann BC for all the boundaries, and I got this,
![image (4)](https://user-images.githubusercontent.com/63679626/216476542-945d083c-e12c-4f50-b8a9-80aa5dd696d4.png)

Then, I use periodic for top and bottom boundary, Neumann for the left and right boundary, I got this,
![image (5)](https://user-images.githubusercontent.com/63679626/216476725-7fdaa0bb-9e9b-42b7-b61a-4299c49fdca7.png)
From the periodic case, the triple junction behaviour in the middle is different from that at the boundaries. 

Could you tell me how is the periodic BC implemented in MOOSE? Where can I get the periodic BC implementation code?
Also, is the 2D model in MOOSE really a 2D model, or, it is actually 3D, but with the third dimension unchanged?

**Reply by GiudGiud** [reply]
The periodic BC is actually implemented in libmesh more than in MOOSE. You can have a look at the doxygen to look at the code but it will use some libmesh objects in there
https://mooseframework.inl.gov/docs/doxygen/moose/classAddPeriodicBCAction.html

The results you are getting look periodic to me
Should they match the neumann BC ? Why ? because of symmetry?

**Reply by abc-hy** [reply]
I still have a question about the dimension of model in MOOSE.
Also, are the 2D models in MOOSE really only have 2D or, it actually has 3D, but with the third dimension unchanged?

**Reply by GiudGiud** [reply]
We have true 2D, with 2D elements, and true 3D with 3D elements

**Reply by abc-hy** [reply]
How does true 2D look like?
![image](https://user-images.githubusercontent.com/63679626/217358744-a2d02db8-3cb6-4934-85ce-82bdcfcbf943.png)
Is this true 2D? without the third axis, z.

**Reply by abc-hy** [reply]
the z here is 0, so this means the z axis does not exist?

**Reply by GiudGiud** [reply, marked as answer]
QUAD4 are 2D eleemnts.

You dont need to specify the z parameters

**Reply by abc-hy** [reply]
How do you know? I did not find the introduction of QUAD4

**Reply by GiudGiud** [reply]
have a look at libmesh doxygen for more info on QUAD4
